astore/client/commands/formatter.go

```diff
@@ -85,21 +90,88 @@ func (ff *TableFormatter) Element(el *astore.Element) {
 	if !ff.elHeaderPrinted {
 		if !ff.disableNesting {
 			if ff.afHeaderPrinted {
-				fmt.Printf("\n")
+				fmt.Fprintf(ff.outputFile, "\n")
 			}
-			ff.tPrint("Children paths\n")
+			ff.tPrint(ff.outputFile, "Children paths\n")
 		}
 
-		fmt.Printf(prefix + "|")
-		ff.hPrint(" %-23s %-30s %-14s\n", "Created", "Creator", "Name")
+		fmt.Fprintf(ff.outputFile, prefix + "|")
+		ff.hPrint(ff.outputFile, " %-23s %-30s %-14s\n", "Created", "Creator", "Name")
 		ff.elHeaderPrinted = true
 	}
 
-	fmt.Printf(prefix+"| %-23s %-30s %-14s\n",
+	fmt.Fprintf(ff.outputFile, prefix+"| %-23s %-30s %-14s\n",
 		time.Unix(0, el.Created).Format("2006-01-02 15:04:05.000"), el.Creator, el.Name)
 }
 
 func (ff *TableFormatter) Flush() {
 	ff.afHeaderPrinted = false
 	ff.elHeaderPrinted = false
 }
+
+type JsonFormatter struct {
+	onePrinted bool
+	outputFile *os.File
+}
+
+func NewJsonFormatter(outputFile *os.File) *JsonFormatter {
+	j := &JsonFormatter{
+		onePrinted: false,
+		outputFile: outputFile,
+	}
+	return j
+}
+
+func (jf *JsonFormatter) Artifact(af *astore.Artifact) {
+	if !jf.onePrinted {
+		// open array of objects
+		fmt.Fprintf(jf.outputFile, "[")
```

Comment:
proposal: rather than print the json manually from here, how about:
1) accumulate in an array / struct
2) on Flush, use "lib/config/marshal" `MarshalFile("name of the file", pointer to struct)`? MarshalToFile will encode in json/yaml/gob/toml based on the file extension, and serialize using an official library (so all escaping etc is done properly).

Reply:
ok